compiler/optimizer/OMRLocalCSE.cpp

```diff
@@ -280,7 +280,7 @@ void OMR::LocalCSE::transformBlock(TR::TreeTop * entryTree, TR::TreeTop * exitTr
    _numNodes = 0;
    _numNullCheckNodes = 0;
    _numCopyPropagations = 0;
-   _arrayRefNodes.deleteAll();
+   _arrayRefNodes = new (stackMemoryRegion) TR_ScratchList<TR::Node>(trMemory());
```

Comment:
How does this lead to a reduction in memory usage ? Would'nt the deleteAll at least keep the list object around ?